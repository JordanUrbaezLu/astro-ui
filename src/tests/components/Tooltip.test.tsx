import {
  fireEvent,
  render,
  screen,
  waitFor,
} from "@testing-library/react";
import {
  Tooltip,
  TooltipPosition,
} from "../../components/Tooltip/Tooltip";
import { axe } from "jest-axe";

describe("Tooltip", () => {
  test("Should render component correctly.", () => {
    const { container } = render(
      <Tooltip
        startsOpen
        text="Tooltip"
        trigger={<button>Button</button>}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe("Accessibility", () => {
  test("Should have no accessibility violations.", async () => {
    const { container } = render(
      <Tooltip
        startsOpen
        text="Tooltip"
        trigger={<button>Button</button>}
      />
    );
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});

describe("Interaction", () => {
  test("Should show tooltip correctly on hover.", async () => {
    render(
      <Tooltip text="Tooltip" trigger={<button>Button</button>} />
    );

    fireEvent.mouseEnter(screen.getByRole("button"));

    await waitFor(() => screen.getByRole("tooltip"));

    expect(screen.getByText("Tooltip")).toBeInTheDocument();

    fireEvent.mouseLeave(screen.getByRole("button"));
  });
});

describe("Props", () => {
  test("Should render text correctly.", () => {
    render(
      <Tooltip
        startsOpen
        text="Tooltip"
        trigger={<button>Button</button>}
      />
    );
    expect(screen.getByText("Tooltip")).toBeInTheDocument();
  });
  test.each<TooltipPosition>(["top", "bottom"])(
    "Should render positions correctly.",
    (position) => {
      render(
        <Tooltip
          startsOpen
          position={position}
          text="Tooltip"
          trigger={<button>Button</button>}
        />
      );
      expect(screen.getByText("Tooltip")).toHaveClass(position);
    }
  );
  test("Should render trigger correctly.", () => {
    render(
      <Tooltip
        startsOpen
        text="Tooltip"
        trigger={<button>Button</button>}
      />
    );
    expect(
      screen.getByRole("button", { name: "Button" })
    ).toBeInTheDocument();
  });
  test("Should render hasNubbin correctly.", () => {
    render(
      <Tooltip
        hasNubbin
        startsOpen
        text="Tooltip"
        trigger={<button>Button</button>}
      />
    );
    expect(screen.getByText("Tooltip")).toHaveClass("nubbin");
  });
  test("Should render startsOpen correctly.", () => {
    render(
      <div data-testid="div">
        <Tooltip
          startsOpen={false}
          text="Tooltip"
          trigger={<button>Button</button>}
        />
      </div>
    );
    expect(screen.getByTestId("div")).not.toHaveTextContent(
      "Tooltip"
    );
  });
});
