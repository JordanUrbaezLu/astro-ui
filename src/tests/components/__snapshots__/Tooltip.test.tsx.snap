// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip Should render component correctly. 1`] = `
<div
  class="container"
>
  <button
    role="button"
    tabindex="0"
  >
    Button
  </button>
  <div
    class="tooltipLayoutContainer"
    style="transform: translate3d(0px, 5px, 0);"
  >
    <div
      class="tooltip bottom"
      role="tooltip"
    >
      Tooltip
    </div>
  </div>
</div>
`;
